fe/fe-core/src/main/java/org/apache/doris/nereids/util/ExpressionUtils.java

```diff
@@ -770,6 +771,8 @@ public static Optional<Expression> checkAndMaybeCommute(Expression expression) {
             return Optional.ofNullable(predicate.left().isSlot() && predicate.right().isLiteral() ? predicate : null);
         } else if (expression instanceof IsNull) {
             return Optional.ofNullable(((IsNull) expression).child().isSlot() ? expression : null);
+        } else if (expression instanceof StringRegexPredicate) {
+            return Optional.ofNullable(expression);
```

Comment:
should not do this change since StringRegexPredicate could not do bucket prune

Reply:
have changed logic. pls help to review.